### PR TITLE
Upgrade to Lucene 9.2.0

### PR DESCRIPTION
**Pull request by mkleen**
https://lucene.apache.org/core/9_2_0/changes/Changes.html

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/testFixtures/java/org/apache/lucene/index/PatchedAssertingLeafReader.java [hunk 1]
```diff
@@ -1078,18 +1083,8 @@ private void assertStats(int maxDoc) {
...
 
         @Override
-        public void intersect(IntersectVisitor visitor) throws IOException {
-            assertThread("Points", creationThread);
-            in.intersect(new AssertingIntersectVisitor(in.getNumDimensions(), in.getNumIndexDimensions(),
-                    in.getBytesPerDimension(), visitor));
-        }
-
-        @Override
-        public long estimatePointCount(IntersectVisitor visitor) {
```
Comment (mkleen) [inline server/src/testFixtures/java/org/apache/lucene/index/PatchedAssertingLeafReader.java:1088]: `estimatePointCount` is final on base class now.

File: server/src/test/java/io/crate/lucene/BitStringQueryTest.java [hunk 2]
```diff
@@ -107,7 +107,7 @@ public void test_eq_on_bits_uses_term_query() throws Exception {
     public void test_is_not_null_on_bit_column_uses_doc_values_field_exists_query() throws Exception {
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/lucene/BitStringQueryTest.java:107]: test name is outdated.

File: server/src/test/java/io/crate/integrationtests/ExplainAnalyzeIntegrationTest.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public void testExplainAnalyzeReportsExecutionTimesOnBothNodesInclQueryBreakdown
                 assertThat(timings, Matchers.hasKey("QueryBreakdown"));
 
                 Map<String, Object> queryBreakdown = ((Map) ((List) timings.get("QueryBreakdown")).get(0));
-                assertThat(queryBreakdown, Matchers.hasEntry("QueryName", "BooleanQuery"));
+                assertThat(queryBreakdown, Matchers.hasEntry(is("QueryName"), is(notNullValue())));
```
[the remark this answers is not in the document]
Comment (mkleen) [reply server/src/test/java/io/crate/integrationtests/ExplainAnalyzeIntegrationTest.java]: Its now a constant score query.

File: server/src/test/java/io/crate/lucene/BitStringQueryTest.java (hunk 2 above)
Comment (mkleen) [reply server/src/test/java/io/crate/lucene/BitStringQueryTest.java:107]: ?

**Review by mfussenegger — APPROVED** [review]

**Review by BaurzhanSakhariev — APPROVED** [review]
👍

**Pull request merged**